src/renderer/redux/actions/subscriptions.js

```diff
@@ -14,5 +21,98 @@ export const doChannelUnsubscribe = (subscription: Subscription) => (dispatch: D
     data: subscription,
   });
 
+export const doCheckSubscriptions = () => (
+  dispatch: Dispatch,
+  getState: () => SubscriptionState
+) => {
+  const checkSubscriptionsTimer = setInterval(
+    () =>
+      selectSubscriptions(getState()).map((subscription: Subscription) =>
+        dispatch(doCheckSubscription(subscription))
+      ),
+    CHECK_SUBSCRIPTIONS_INTERVAL
+  );
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTIONS_SUBSCRIBE,
+    data: { checkSubscriptionsTimer },
+  });
+};
+
+export const doCheckSubscription = (subscription: Subscription) => (dispatch: Dispatch) => {
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTION_STARTED,
+    data: subscription,
+  });
+
+  Lbry.claim_list_by_channel({ uri: subscription.uri, page: 1 }).then(result => {
+    const claimResult = result[subscription.uri] || {};
+    const { claims_in_channel: claimsInChannel } = claimResult;
+
+    let count = subscription.latest
+      ? claimsInChannel.reduce(
+          (prev, cur, index) =>
+            buildURI({ contentName: cur.name, claimId: cur.claim_id}, false) === subscription.latest ? index : prev,
+          -1
+        )
+      : 1;
+
+    if (count !== 0) {
+      if (!claimsInChannel[0].value.stream.metadata.fee) {
+        dispatch(
+          doPurchaseUri(buildURI({ contentName: claimsInChannel[0].name, claimId: claimsInChannel[0].claim_id }, false), { cost: 0 })
+        );
+      }
+
+      const notif = new window.Notification(subscription.channelName, {
```

Comment:
Can we wait to show this notification until after content is ready to watch? I.e. after `get` has successfully started? That way when the user clicks the notification they get something immediately.

Reply:
Yes, @seanyesmunt brought this up as well. I have reasons for why I didn't make it this way, which I think we should discuss -- although I think you are right that we probably want to download it first and I probably should have implemented it this way. I will schedule this in the next sprint.

Reply:
I checked out the code here:
1) `doLoadVideo` and `doPurchaseUri` are confusingly -- almost backwardly -- named. It may make sense to tweak these.
2) Since you are already checking that the price is free, it seems like you could potentially skip `doPurchaseUri` and go right to `doLoadVideo`.
3) Both `doLoadVideo` and `doPurchaseUri` can trigger modal popups. It is weird that asynchronous background loading of content can trigger a modal. Fairly guaranteed that this will cause confusion at some point.
4) It doesn't seem particularly tricky to attach a notification to the success of the call to `get`. I think a light modification of `doLoadVideo` to take a notification or callback would get you there.